trillian/ctfe/handlers.go

```diff
@@ -123,8 +123,8 @@ type PathHandlers map[string]AppHandler
 // AppHandler holds a LogContext and a handler function that uses it, and is
 // an implementation of the http.Handler interface.
 type AppHandler struct {
-	Context LogContext
-	Handler func(context.Context, LogContext, http.ResponseWriter, *http.Request) (int, error)
+	Context *LogContext
```

Comment:
Tempted to quibble about opening this up for random mutations but I think the impact is limited as it's only really accessible to handlers and wrappers. I think maybe LogContext should be split up more but it's an arguable point.